### PR TITLE
Dev fix php stan commonobject

### PR DESCRIPTION
**Pull request by moreauftheobald**
# FIX
Fix PHP Stan Error on commonobject.class.php

Doxygene stat:
htdocs/core/class/commonobject.class.php	9569	Access to an undefined property CommonObject::$pass.
htdocs/core/class/commonobject.class.php	10121	Call to an undefined method CommonObject::fetch().
htdocs/core/class/commonobject.class.php	10129	Call to an undefined method CommonObject::delete().

### CODE REVIEW

File: htdocs/bom/class/bom.class.php [hunk 1]
```diff
@@ -1806,6 +1785,7 @@ public function __construct(DoliDB $db)
 		global $conf, $langs;
 
 		$this->db = $db;
+		$this->setPicto('bomline');
```
Comment (eldy) [inline htdocs/bom/class/bom.class.php:1788]: What is the goal to have this compared to old code ?

**Comment by eldy** [comment]
The pr seems to contain a lot of change that is related to refactoring (not allowed during a beta) and not to fix warning (allowed during a beta)

File: htdocs/bom/class/bom.class.php (hunk 1 above)
Comment (moreauftheobald) [reply htdocs/bom/class/bom.class.php:1788]: regarding to PHP 8.2 documentation it seems to be the best way to not overrid the property $picto of the common object line .
if in the __construct method you just call $this->picto = 'bomline' PHP consider this as on override.

**Comment by moreauftheobald** [comment]
error from my self this PR mix to PR.... i close it and send a new one ;)

**Comment by moreauftheobald** [comment]
See #26879

**Pull request closed without merging**